plugins/connection/kubectl.py

```diff
@@ -181,6 +181,83 @@
         aliases: [ kubectl_verify_ssl ]
 """
 
+EXAMPLES = r"""
+
+- name: Run a command in a pod using local kubectl with kubeconfig file ~/.kube/config
```

Comment:
This example does not work as expected. The task runs on localhost. I suspect this has to do with [implicit localhost](https://docs.ansible.com/ansible/latest/inventory/implicit_localhost.html).

Reply:
yep, you are right, I make mistake and it doesn't work. Instead of `connection: kubernetes.core.kubectl` should be used variable `ansible_connection: kubernetes.core.kubectl`, so, following works fine:

```
root@python:/# cat <<EOF > test.yaml
> ---
> - name: Run local commands
>   hosts: localhost
>   gather_facts: no
>   tasks:
>     - name: Check the os-release content
>       ansible.builtin.command: cat /etc/os-release
>       register: local_os_release
>     - name: check kubeclt and cluster apiVersion
>       ansible.builtin.command: kubectl version
>
> - name: Run a command in a pod using local kubectl with kubeconfig file ~/.kube/config
>   hosts: localhost
>   gather_facts: no
>   vars:
>     ansible_connection: kubernetes.core.kubectl
>     ansible_kubectl_namespace: test
>     ansible_kubectl_pod: ubuntu-01
>   tasks:
>     - name: Run a command in a pod
>       ansible.builtin.command: cat /etc/os-release
>       register: pod_os_release
>
>     - debug:
>         msg: "{{ pod_os_release.stdout_lines }} is not the same as {{ local_os_release.stdout_lines }}"
>       when: pod_os_release.stdout_lines != local_os_release.stdout_lines
> EOF
root@python:/# ansible-playbook test.yaml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Run local commands] *****************************************************************************************************************************************************************************************

TASK [Check the os-release content] *******************************************************************************************************************************************************************************
changed: [localhost]

TASK [check kubeclt and cluster apiVersion] ***********************************************************************************************************************************************************************
changed: [localhost]

PLAY [Run a command in a pod using local kubectl with kubeconfig file ~/.kube/config] *****************************************************************************************************************************

TASK [Run a command in a pod] *************************************************************************************************************************************************************************************
changed: [localhost]

TASK [debug] ******************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "['PRETTY_NAME=\"Ubuntu 24.04 LTS\"', 'NAME=\"Ubuntu\"', 'VERSION_ID=\"24.04\"', 'VERSION=\"24.04 LTS (Noble Numbat)\"', 'VERSION_CODENAME=noble', 'ID=ubuntu', 'ID_LIKE=debian', 'HOME_URL=\"https://www.ubuntu.com/\"', 'SUPPORT_URL=\"https://help.ubuntu.com/\"', 'BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\"', 'PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\"', 'UBUNTU_CODENAME=noble', 'LOGO=ubuntu-logo'] is not the same as ['PRETTY_NAME=\"Debian GNU/Linux 12 (bookworm)\"', 'NAME=\"Debian GNU/Linux\"', 'VERSION_ID=\"12\"', 'VERSION=\"12 (bookworm)\"', 'VERSION_CODENAME=bookworm', 'ID=debian', 'HOME_URL=\"https://www.debian.org/\"', 'SUPPORT_URL=\"https://www.debian.org/support\"', 'BUG_REPORT_URL=\"https://bugs.debian.org/\"']"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhroot@python:/# cat <<EOF > test.yaml
> ---
> - name: Run local commands
>   hosts: localhost
>   gather_facts: no
>   tasks:
>     - name: Check the os-release content
>       ansible.builtin.command: cat /etc/os-release
>       register: local_os_release
>     - name: check kubeclt and cluster apiVersion
>       ansible.builtin.command: kubectl version
>
> - name: Run a command in a pod using local kubectl with kubeconfig file ~/.kube/config
>   hosts: localhost
>   gather_facts: no
>   vars:
>     ansible_connection: kubernetes.core.kubectl
>     ansible_kubectl_namespace: test
>     ansible_kubectl_pod: ubuntu-01
>   tasks:
>     - name: Run a command in a pod
>       ansible.builtin.command: cat /etc/os-release
>       register: pod_os_release
>
>     - debug:
>         msg: "{{ pod_os_release.stdout_lines }} is not the same as {{ local_os_release.stdout_lines }}"
>       when: pod_os_release.stdout_lines != local_os_release.stdout_lines
> EOF
root@python:/# ansible-playbook test.yaml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Run local commands] *****************************************************************************************************************************************************************************************

TASK [Check the os-release content] *******************************************************************************************************************************************************************************
changed: [localhost]

TASK [check kubeclt and cluster apiVersion] ***********************************************************************************************************************************************************************
changed: [localhost]

PLAY [Run a command in a pod using local kubectl with kubeconfig file ~/.kube/config] *****************************************************************************************************************************

TASK [Run a command in a pod] *************************************************************************************************************************************************************************************
changed: [localhost]

TASK [debug] ******************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "['PRETTY_NAME=\"Ubuntu 24.04 LTS\"', 'NAME=\"Ubuntu\"', 'VERSION_ID=\"24.04\"', 'VERSION=\"24.04 LTS (Noble Numbat)\"', 'VERSION_CODENAME=noble', 'ID=ubuntu', 'ID_LIKE=debian', 'HOME_URL=\"https://www.ubuntu.com/\"', 'SUPPORT_URL=\"https://help.ubuntu.com/\"', 'BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\"', 'PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\"', 'UBUNTU_CODENAME=noble', 'LOGO=ubuntu-logo'] is not the same as ['PRETTY_NAME=\"Debian GNU/Linux 12 (bookworm)\"', 'NAME=\"Debian GNU/Linux\"', 'VERSION_ID=\"12\"', 'VERSION=\"12 (bookworm)\"', 'VERSION_CODENAME=bookworm', 'ID=debian', 'HOME_URL=\"https://www.debian.org/\"', 'SUPPORT_URL=\"https://www.debian.org/support\"', 'BUG_REPORT_URL=\"https://bugs.debian.org/\"']"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

root@python:/#ost                  : ok=4    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

root@python:/# ansible-galaxy collection list

# /root/.ansible/collections/ansible_collections
Collection      Version
--------------- -------
kubernetes.core 3.1.0
root@python:/# ansible --version
ansible [core 2.16.7]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.12/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.12.3 (main, May 14 2024, 07:34:56) [GCC 12.2.0] (/usr/local/bin/python)
  jinja version = 3.1.4
  libyaml = True
root@python:/#
```

I will retest all examples and update PR

Reply:
Done, thanks